### PR TITLE
Switch over python print calls for logging calls

### PR DESCRIPTION
**Pull request by ChaoticTempest**
Addresses #2780. This swaps `print` statements into `logger.info` statements. Note that I also switched over the print statements for python scripts inside the `scripts/` folder. This required a bit of some pathing magic to import the logger from the `pytests` folder since the python files are spread across the whole project without `__init__.py` for proper modules.

### CODE REVIEW

File: pytest/lib/mocknet.py [hunk 1]
```diff
@@ -83,7 +85,7 @@ def list_validators(node):
 
 def setup_python_environment(node, wasm_contract):
     m = node.machine
-    print(f'INFO: Setting up python environment on {m.name}')
+    logger.info(f'INFO: Setting up python environment on {m.name}')
```
Comment (ailisp) [inline pytest/lib/mocknet.py:88]: I guess `INFO` can be removed since logger.info will tell?

File: pytest/lib/cluster.py [hunk 2]
```diff
@@ -206,7 +208,7 @@ def validators(self):
                 self.get_status()['validators']))
 
     def stop_checking_store(self):
-        print("WARN: Stopping checking Storage for inconsistency for %s:%s" %
+        logger.info("WARN: Stopping checking Storage for inconsistency for %s:%s" %
```
Comment (ailisp) [inline pytest/lib/cluster.py:211]: Also this, use `logger.warn("Stopping checking Storage ...")` instead?

File: pytest/lib/cluster.py [hunk 3]
```diff
@@ -218,7 +220,7 @@ def check_store(self):
                 pass
             else:
                 if res['result'] == 0:
-                    print(
+                    logger.info(
```
Comment (ailisp) [inline pytest/lib/cluster.py:223]: logger.error

File: scripts/state/update_res.py [hunk 4]
```diff
@@ -9,14 +9,18 @@
...
+
 
 def main():
     if len(sys.argv) == 1:
         update_res()
     elif len(sys.argv) == 2 and sys.argv[1] == 'check':
         check_res()
     else:
-        print('Usage: update-res.py | update-res.py check')
+        logger.info('Usage: update-res.py | update-res.py check')
```
Comment (ailisp) [inline scripts/state/update_res.py:23]: This probably should use `print` or print to stderr

**Review by ailisp — APPROVED** [review]
Overall looks good, some logging can be improved

**Review by bowenwang1996 — COMMENTED** [review]
Please run nayduck to make sure that this change does not accidentally break anything

**Comment by ChaoticTempest** [comment]
So, I changed some strings from using `%s` to an f-string instead, but that throws invalid syntax since the script itself is using python2 via `#!/usr/bin/env python`. But some other scripts use python3. The mixing of python versions will be a potential issue down the line. Do you guys know of any problems that would arise from switching all of scripts over to `#!/usr/bin/env python3`?

File: pytest/lib/mocknet.py (hunk 1 above)
Comment (ChaoticTempest) [reply pytest/lib/mocknet.py:88]: Currently, I don't have the logger formatted to show the what the log level is at. I can added it if necessary, but would also end up being added to every other log as well that didn't have `INFO:` prefixed before

**Comment by chefsale** [comment]
Please move them all to Python 3.

File: pytest/lib/branches.py [hunk 5]
```diff
@@ -120,6 +121,6 @@ def prepare_ab_test(other_branch):
         if not os.getenv('NAYDUCK'):
             compile_binary(str(other_branch))
         else:
-            print('RC binary should be downloaded for NayDuck.')
+            logger.info('RC binary should be downloaded for NayDuck.')
```
Comment (mina86) [inline pytest/lib/branches.py:124]: `logging.critical`. After all, we’re aborting the script.

Here and throughout the commit where we log before aborting with a `sys.exit(1)`.

File: pytest/lib/cluster.py [hunk 6]
```diff
@@ -508,10 +510,10 @@ def reset_validator_key(self, new_key):
...
 
 def github_auth():
-    print("Go to the following link in your browser:")
-    print()
-    print("http://nayduck.eastus.cloudapp.azure.com:3000/local_auth")
-    print()
+    logger.info("Go to the following link in your browser:")
+    logger.info("")
+    logger.info("http://nayduck.eastus.cloudapp.azure.com:3000/local_auth")
+    logger.info("")
```
Comment (mina86) [inline pytest/lib/cluster.py:516]: Like before, this is an interactive function. I don’t think this should be converted to use logger.

File: pytest/lib/network.py [hunk 7]
```diff
@@ -16,17 +18,17 @@ def init_network_pillager():
...
                 "Make sure the current user has access to it, e.g. by changing the owner:"
             )
-            print("")
-            print(
+            logger.info("")
+            logger.info(
                 "    chown <group>.<user> /sys/fs/cgroup/net_cls/block/net_cls.classid"
             )
-            print("")
+            logger.info("")
```
Comment (mina86) [inline pytest/lib/network.py:31]: `logger.critical`

File: pytest/lib/configured_logger.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        handler = logging.StreamHandler(sys.stdout)
+        handler.setLevel(level)
+    log.addHandler(handler)
+    log.propagate = False
+
+    return log
+
+
+# global logger for testing purposes:
+logger = new_logger("test")
```
Comment (mina86) [inline pytest/lib/configured_logger.py:46]: `test_logger` perhaps? Not sure though; maybe `logger` is fine.

File: pytest/lib/bridge.py [hunk 9]
```diff
@@ -175,9 +175,9 @@ def cleanup(self):
                 self.stderr.close()
             self.kill()
         except BaseException:
-            print("Kill %s failed on cleanup!" % (self.__class__.__name__))
+            logger.info("Kill %s failed on cleanup!" % (self.__class__.__name__))
             traceback.print_exc()
```
Comment (mina86) [inline pytest/lib/bridge.py:179]: logger.error('Kill %s failed on cleanup', type(self).__name__, exc_info=sys.exc_info())

No need for `traceback.print_exc()` or the two empty lines.

File: pytest/lib/bridge.py [hunk 10]
```diff
@@ -137,11 +137,11 @@ def __init__(self,
         self.custom = dict()
 
 def atexit_cleanup(obj):
-    print("Cleaning %s on script exit" % (obj.__class__.__name__))
+    logger.info("Cleaning %s on script exit" % (obj.__class__.__name__))
```
Comment (mina86) [inline pytest/lib/bridge.py:140]: logging does %-formatting so this would be better written as:

    logger.info('Cleaning %s on script exit', type(obj).__name__)

Here and throughout.

File: pytest/tests/bridge/end2end.py [hunk 11]
```diff
@@ -4,9 +4,10 @@
 # If `no_txs_in_parallel`, no txs will be executed in parallel.
 
 import sys, time
+from configured_logger import logger
 
 if len(sys.argv) < 3:
-    print("python end2end.py <eth2near_tx_number> <near2eth_tx_number> [...]")
+    logger.info("python end2end.py <eth2near_tx_number> <near2eth_tx_number> [...]")
```
Comment (mina86) [inline pytest/tests/bridge/end2end.py:10]: This should stay print.

File: pytest/lib/test.py [hunk 12]
```diff
@@ -0,0 +1,6 @@
+
+
+from configured_logger import new_logger
+
+L = new_logger(outfile="/tmp/something.txt")
+L.info("womp")
```
Comment (mina86) [inline pytest/lib/test.py:6]: Was this file intended to be included? Same with `welp.txt`?

File: pytest/tests/mocknet/load_testing.py [hunk 13]
```diff
@@ -81,79 +83,79 @@ def check_tps(measurement, expected_in, expected_out=None, tolarance=0.05):
...
     time.sleep(5)
-    print('INFO: All transaction types results:')
+    logger.info('INFO: All transaction types results:')
     all_tx_measurement = measure_tps_bps(nodes)
     test_passed = (all_tx_measurement['bps'] > 0.5) and test_passed
     test_passed = check_memory_usage(nodes[0]) and test_passed
     test_passed = check_slow_blocks(all_tx_initial_metrics,
                                     final_metrics) and test_passed
-    print('')
+    logger.info('')
```
Comment (mina86) [inline pytest/tests/mocknet/load_testing.py:154]: I think we should be dropping empty log lines.

File: pytest/lib/cluster.py [hunk 14]
```diff
@@ -361,9 +363,9 @@ def cleanup(self):
         try:
             self.kill()
         except:
-            print("Kill failed on cleanup!")
+            logger.info("Kill failed on cleanup!")
             traceback.print_exc()
-            print("\n\n")
+            logger.info("\n\n")
```
Comment (mina86) [inline pytest/lib/cluster.py:368]: logger.critical('Kill failed on cleanup!', exc_info=sys.exc_info())

should do the trick, no?

File: pytest/tests/sanity/lightclnt.py [hunk 15]
```diff
@@ -96,7 +97,7 @@ def get_up_to(from_, to):
 
     for i in range(from_, to + 1):
         hash_ = height_to_hash[i]
-        print(i, hash_, hash_to_epoch[hash_], hash_to_next_epoch[hash_])
+        logger.info(i, hash_, hash_to_epoch[hash_], hash_to_next_epoch[hash_])
```
Comment (mina86) [inline pytest/tests/sanity/lightclnt.py:100]: This needs to be changed to use format string. Here and below in this file.

File: pytest/tests/sanity/proxy_simple.py [hunk 16]
```diff
@@ -19,9 +20,9 @@ class Handler(ProxyHandler):
     async def handle(self, msg, fr, to):
         if msg.enum == 'Block':
             h = msg.Block.BlockV2.header.inner_lite().height
-            print("Height:", h)
+            logger.info("Height: {h}")
```
Comment (mina86) [inline pytest/tests/sanity/proxy_simple.py:23]: You’ve missed `f`.

**Comment by mina86** [comment]
> Note that I also switched over the print statements for python scripts inside the scripts/ folder.

I’d argue those should be left alone. They are interactive tools after all.

> So, I changed some strings from using %s to an f-string instead,

Wouldn’t %-formatting be more idiomatic for logging though? I mean, logging performs the formatting if it’s called with multiple arguments as in `logging.info('foo %s baz', 'bar')`.

File: pytest/lib/cluster.py (hunk 6 above)
Comment (chefsale) [reply pytest/lib/cluster.py:516]: As mentioned below by @mina86 , we can also drop the `logger.info("")`

File: nightly/nayduck.py [hunk 17]
```diff
@@ -69,10 +73,10 @@ def get_tests(fl):
...
 
 def github_auth():
-    print("Go to the following link in your browser:")
-    print()
-    print("http://nayduck.eastus.cloudapp.azure.com:3000/local_auth")
-    print()
+    logger.info(
+        "Go to the following link in your browser:\n"
+        "http://nayduck.eastus.cloudapp.azure.com:3000/local_auth\n"
+    )
```
[the remark this answers is not in the document]
Comment (ChaoticTempest) [reply nightly/nayduck.py:79]: I'll just revert this file then

File: pytest/lib/test.py (hunk 12 above)
Comment (ChaoticTempest) [reply pytest/lib/test.py:6]: Good catch. Accidentally committed these

File: pytest/tests/sanity/lightclnt.py (hunk 15 above)
Comment (ChaoticTempest) [reply pytest/tests/sanity/lightclnt.py:100]: Wow, I thought I got all these with my regex matching, but guess not

Comment (mina86) [reply pytest/tests/sanity/lightclnt.py:100]: https://xkcd.com/1171/

**Comment by ChaoticTempest** [comment]
> I’d argue those should be left alone. They are interactive tools after all.

Good point. I just ported them over since they were the same when I was just outputting them to stdout, but they also include timestamps now.

> Wouldn’t %-formatting be more idiomatic for logging though? I mean, logging performs the formatting if it’s called with multiple arguments as in `logging.info('foo %s baz', 'bar')`.

I don't think it's idiomatic though, since it's still using `msg % args` underneath which still has some gotchas. Quote from the python docs:
> The formatting operations described here exhibit a variety of quirks that lead to a number of common errors (such as failing to display tuples and dictionaries correctly). Using the newer formatted string literals, the str.format() interface, or template strings may help avoid these errors.

File: pytest/lib/bridge.py (hunk 10 above)
Comment (ChaoticTempest) [reply pytest/lib/bridge.py:140]: I think it'd be consistent to just change these all over to just f-strings and not have all the gotchas I mentioned, but was trying not to get this ticket out of scope by doing that. Since this ticket was to just change over to logging

**Comment by ChaoticTempest** [comment]
@chefsale @mina86 So what do you guys think. Should I just revert the `scripts/` folder back to using `print` since its meant to be interactive, but the logging still outputs to stdout either ways. Or we can cherrypick the most interactive pieces and just have them be in print calls instead

**Comment by chefsale** [comment]
I think it is fine to revert the scripts and leave the rest. We eitherways have to delete some of the scriots there as they are obsolete. I will take a look at that onxe we land this one.

**Comment by ChaoticTempest** [comment]
@chefsale Alright sounds good. It's reverted

**Review by chefsale — APPROVED** [review]
This is an improvement, so LGTM.
